File: wrappers/java/src/test/java/org/hyperledger/indy/sdk/crypto/CreateKeyTest.java
package org.hyperledger.indy.sdk.crypto;

import org.bitcoinj.core.Base58;
import org.hyperledger.indy.sdk.IndyIntegrationTestWithSingleWallet;
import org.hyperledger.indy.sdk.InvalidStructureException;
import org.junit.Test;

import java.util.concurrent.ExecutionException;

import static org.hamcrest.CoreMatchers.isA;
import static org.junit.Assert.assertEquals;


public class CreateKeyTest extends IndyIntegrationTestWithSingleWallet {

	@Test
	public void testCreateKeyWorksForSeed() throws Exception {
		String senderVk = Crypto.createKey(wallet, MY1_IDENTITY_KEY_JSON).get();
		assertEquals(32, Base58.decode(senderVk).length);
	}

	@Test
	public void testCreateKeyWorksWithoutSeed() throws Exception {
		String senderVk = Crypto.createKey(wallet, "{}").get();
		assertEquals(32, Base58.decode(senderVk).length);
	}

	@Test
	public void testCreateKeyWorksForInvalidSeed() throws Exception {
		String paramJson = new CryptoJSONParameters.CreateKeyJSONParameter("invalidSeedLength", null).toJson();

		thrown.expect(ExecutionException.class);
		thrown.expectCause(isA(InvalidStructureException.class));

		Crypto.createKey(wallet, paramJson).get();
	}
}

File: wrappers/dotnet/indy-sdk-dotnet-test/CryptoTests/CreateKeyTest.cs
using Hyperledger.Indy.CryptoApi;
using Hyperledger.Indy.Test.Util.Base58Check;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System.Threading.Tasks;

namespace Hyperledger.Indy.Test.CryptoTests
{
    [TestClass]
    public class CreateKeyTest : IndyIntegrationTestWithSingleWallet
    {
        [TestMethod]
        public async Task TestCreateKeyWorksForSeed()
        {
            var senderVk = await Crypto.CreateKeyAsync(wallet, MY1_IDENTITY_KEY_JSON);
            Assert.AreEqual(32, Base58CheckEncoding.DecodePlain(senderVk).Length);
        }

        [TestMethod]
        public async Task TestCreateKeyWorksWithoutSeed()
        {
            var senderVk = await Crypto.CreateKeyAsync(wallet, "{}");
            Assert.AreEqual(32, Base58CheckEncoding.DecodePlain(senderVk).Length);
        }

        [TestMethod]
        public async Task TestCreateKeyWorksForInvalidSeed()
        {
            var paramJson = string.Format("{{\"seed\":\"{0}\"}}", "invalidSeedLength");

            var ex = await Assert.ThrowsExceptionAsync<InvalidStructureException>(() =>
               Crypto.CreateKeyAsync(wallet, paramJson)
           );
        }
    }
}
